pkg/cmd/cli/cmd/rsh.go

```diff
@@ -155,8 +155,9 @@ func (o *RshOptions) Validate() error {
 // Run starts a remote shell session on the server
 func (o *RshOptions) Run() error {
 	// Insert the TERM into the command to be run
-	term := fmt.Sprintf("TERM=%s", util.Env("TERM", "xterm"))
-	o.Command = append([]string{"env", term}, o.Command...)
-
+	if len(o.Command) == 1 && o.Command[0] == o.Executable {
+		termsh := fmt.Sprintf("TERM=%q %s", util.Env("TERM", "xterm"), o.Executable)
+		o.Command = append(o.Command, "-c", termsh)
```

Comment:
my only fear now is that since we're doing this for any value of --shell the user passes, if they pass a --shell value that doesn't support "-c", things will be broken.  That was part of my motivation to only do this logic when the shell was explicitly "/bin/sh".